backpack/extensions/firstorder/sum_grad_squared/sgs_base.py

```diff
@@ -1,20 +1,71 @@
+"""Contains SGSBase, the base module for sum_grad_squared extension."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable, List, Tuple
+
+from torch import Tensor
+from torch.nn import Module
+
+from backpack.core.derivatives.basederivatives import BaseParameterDerivatives
 from backpack.extensions.firstorder.base import FirstOrderModuleExtension
 
+if TYPE_CHECKING:
+    from backpack.extensions import SumGradSquared
+
 
 class SGSBase(FirstOrderModuleExtension):
-    def __init__(self, derivatives, params=None):
-        self.derivatives = derivatives
-        self.N_axis = 0
-        super().__init__(params=params)
-
-    def bias(self, ext, module, g_inp, g_out, bpQuantities):
-        grad_batch = self.derivatives.bias_jac_t_mat_prod(
-            module, g_inp, g_out, g_out[0], sum_batch=False
-        )
-        return (grad_batch ** 2).sum(self.N_axis)
-
-    def weight(self, ext, module, g_inp, g_out, bpQuantities):
-        grad_batch = self.derivatives.weight_jac_t_mat_prod(
-            module, g_inp, g_out, g_out[0], sum_batch=False
-        )
-        return (grad_batch ** 2).sum(self.N_axis)
+    """Base class for extensions calculating sum_grad_squared."""
+
+    def __init__(self, derivatives: BaseParameterDerivatives, params: List[str] = None):
+        """Initialization.
+
+        For each parameter a function is initialized that is named like the parameter
+
+        Args:
+            derivatives: calculates the derivatives wrt parameters
+            params: list of parameter names
+        """
+        self.derivatives: BaseParameterDerivatives = derivatives
+        self.N_axis: int = 0
```

Comment:
@schaefertim Are these two annotations necessary? (`BaseParameterDerivatives` already in constructor, `0` is `int`)